openapi.yaml

```diff
@@ -143,374 +82,417 @@ paths:
-          required: true
-          type: string
-        - $ref: '#/parameters/osuID'
-        - $ref: '#/parameters/authorization'
+        - $ref: '#/parameters/osuId'
+        - $ref: '#/parameters/mealPlanId'
       responses:
-        200:
+        '200':
           description: Successful response
           schema:
-            $ref: '#/definitions/MealPlanResultObject'
-        404:
-          description: Person or meal plan not found
-          schema:
-            $ref: '#/definitions/Error'
-  /persons/{osuID}/addresses:
+            $ref: '#/definitions/MealPlanResult'
+        '404':
+          $ref: '#/responses/404'
+        '500':
+          $ref: '#/responses/500'
+  /persons/{osuId}/address:
     get:
       description: Get addresses for a person
       tags:
         - addresses
       operationId: getAddresses
       parameters:
+        - $ref: '#/parameters/osuId'
+        - name: city
+          in: query
+          type: string
+          required: false
+          description: Filter results by city.
+        - name: county
+          in: query
+          type: string
+          required: false
+          description: Filter results by county.
+        - name: state
+          in: query
+          type: string
+          required: false
+          description: Filter results by state.
+        - name: stateCode
+          in: query
+          type: string
+          required: false
+          description: Filter results by state code.
+        - name: nation
+          in: query
+          type: string
+          required: false
+          description: Filter results by nation.
+        - name: nationCode
+          in: query
+          type: string
+          required: false
+          description: Filter results by nation code.
         - $ref: '#/parameters/addressType'
-        - $ref: '#/parameters/osuID'
-        - $ref: '#/parameters/authorization'
       responses:
-        200:
+        '200':
           description: Successful response
           schema:
-            $ref: '#/definitions/AddressesResultObject'
-        404:
-          description: Person not found
-          schema:
-            $ref: '#/definitions/Error'
+            $ref: '#/definitions/AddressSetResult'
+        '404':
+          $ref: '#/responses/404'
+        '500':
+          $ref: '#/responses/500'
     post:
       description: Create address for a person
       tags:
         - addresses
       operationId: postAddresses
       parameters:
-        - name: addressBody
+        - name: body
           in: body
-          description: Post body for the new address
+          description: Post body for new address
           required: true
           schema:
-            $ref: '#/definitions/NewAddressResultObject'
-        - $ref: '#/parameters/osuID'
-        - $ref: '#/parameters/authorization'
+            $ref: '#/definitions/AddressPostBody'
+        - $ref: '#/parameters/osuId'
       responses:
-        200:
+        '200':
```

Comment:
Not sure, but should we use 202 for the successful response here?

Reply:
It is 200 in v1. I'm not sure how to check what it should be.

Reply:
All other POST and PATCH/PUT for this API are using 202 accept for successful responses because we need to call some SPs to process the data and it may take times until the data is really updated/inserted, I feel it'll be more consistent to use the same status code here as well. However, since this endpoint is actually calling an SP that won't need extra time to process, in other words, if we didn't get a thrown error, the new data is actually inserted. That's why 200 also sounds fair to me.

Reply:
I agree I'll keep it 200